libstuff/libstuff.cpp

```diff
@@ -161,19 +179,17 @@ void SSyslogSocketDirect(int priority, const char *format, ...) {
     int socketError = 0;
     static const size_t MAX_MESSAGE_SIZE = 8 * 1024;
 
-    // This block shouldn't be required, but it kicks in if this is used from somewhere that fails to call
-    // `SInitialize`. I figured this is better than just losing the log.
-    if (SSyslogSocketFD == 0) {
-        SSyslogSocketAddr.sun_family = AF_UNIX;
-        strcpy(SSyslogSocketAddr.sun_path, "/run/systemd/journal/syslog");
-        SSyslogSocketFD = -1;
-    }
+    // Choose an FD from our array.
+    size_t socketIndex = (++SLogSocketCurrentOffset) % S_LOG_SOCKET_MAX;
```

Comment:
CPP noob here, so maybe I am wrong - wouldn't we want `SLogSocketCurrentOffset++` so we don't skip index 0?

Reply:
we will eventually use 0, this is a modulo, so when `SLogSocketCurrentOffset` is divisible by 500, it will return 0.